### PR TITLE
WIP:  Adding CHECK_DEBIAN 

### PR DESCRIPTION
**Pull request by nim65s**
This PR follows discussions in #167, and should not be merged before it, and before a rebase.

### CODE REVIEW

File: base.cmake [hunk 1]
```diff
@@ -254,3 +254,23 @@ MACRO(SETUP_PROJECT_FINALIZE)
...
+# ---------
+#
+# .. variable:: IS_DEBIAN
+#
+#   **REQUIRED**
+#
+#   A Boolean, indicating if the current system is Debian based
+#
+MACRO(CHECK_DEBIAN IS_DEBIAN)
+  SET(${IS_DEBIAN} EXISTS "/etc/debian_version")
```
Comment (jcarpent) [inline base.cmake]: You should PARENT_SCOPE here.

File: base.cmake [hunk 2]
```diff
@@ -254,3 +254,23 @@ MACRO(SETUP_PROJECT_FINALIZE)
...
+# Variables
+# ---------
+#
+# .. variable:: IS_DEBIAN
+#
+#   **REQUIRED**
+#
+#   A Boolean, indicating if the current system is Debian based
+#
+MACRO(CHECK_DEBIAN IS_DEBIAN)
```
Comment (jcarpent) [inline base.cmake]: I think check_debian should not take any argument but should define a global variable.
You can have a look on https://gist.github.com/lbaehren/2413369

File: base.cmake [hunk 3]
```diff
@@ -261,20 +261,9 @@ ENDMACRO(SETUP_PROJECT_FINALIZE)
...
-#   A Boolean, indicating if the current system is Debian based
-#
-MACRO(CHECK_DEBIAN IS_DEBIAN)
-  IF(EXISTS "/etc/debian_version")
-    SET(IS_DEBIAN TRUE)
-  ELSE(EXISTS "/etc/debian_version")
-    SET(IS_DEBIAN FALSE)
-  ENDIF(EXISTS "/etc/debian_version")
-ENDMACRO(CHECK_DEBIAN IS_DEBIAN)
+MACRO(CHECK_DEBIAN)
```
Comment (jcarpent) [inline base.cmake]: I think you have to add PARENT_SCOPE to a variable to make it public for the whole project.

Comment (nim65s) [reply base.cmake]: I tried this version on a debian system and on another unix system, and `DEBIAN_FOUND` was correct

Comment (nim65s) [reply base.cmake]: Moreover, `PARENT_SCOPE` is not documented in [`FIND_FILE`](https://cmake.org/cmake/help/latest/command/find_file.html)

Comment (jcarpent) [reply base.cmake]: PARENT_SCOPE is when you SET a variable.
Apparently, FIND_FILE is doing that for you.

**Comment by jcarpent** [comment]
I let @bchretien doing the merge.

**Comment by gergondet** [comment]
@nim65s If I'm not mistaken #168 actually superseeds this PR right? Can we close it?

**Pull request merged**